File: app/src/main/java/com/example/josycom/flowoverstack/repository/AnswerRepository.java
package com.example.josycom.flowoverstack.repository;

import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.josycom.flowoverstack.model.Answer;
import com.example.josycom.flowoverstack.model.AnswerResponse;
import com.example.josycom.flowoverstack.network.ApiService;
import com.example.josycom.flowoverstack.network.RestApiClient;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class AnswerRepository {

    private final int questionId;
    private final String order;
    private final String sortCondition;
    private final String site;
    private final String filter;
    private MutableLiveData<List<Answer>> mAnswers = new MutableLiveData<>();

    public AnswerRepository(int questionId, String order, String sortCondition, String site, String filter){
        this.questionId = questionId;
        this.order = order;
        this.sortCondition = sortCondition;
        this.site = site;
        this.filter = filter;
        getAnswersToQuestion();
    }

    private void getAnswersToQuestion() {
        ApiService apiService = RestApiClient.getApiService(ApiService.class);
        Call<AnswerResponse> call = apiService.getAnswersToQuestion(questionId, order, sortCondition, site, filter);
        call.enqueue(new Callback<AnswerResponse>() {
            @Override
            public void onResponse(@NotNull Call<AnswerResponse> call, @NotNull Response<AnswerResponse> response) {
                AnswerResponse answerResponse = response.body();
                if (answerResponse != null) {
                    mAnswers.setValue(answerResponse.getItems());
                } else {
                    Log.d("AnswerRepository", "No answer yet");
                }
            }

            @Override
            public void onFailure(@NotNull Call<AnswerResponse> call, @NotNull Throwable t) {
                t.printStackTrace();
            }
        });
    }

    public LiveData<List<Answer>> getAnswers() {
        return mAnswers;
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/data/AnswerRepository.kt
package com.josycom.mayorjay.flowoverstack.data

import androidx.paging.Pager
import androidx.paging.PagingConfig
import androidx.paging.PagingData
import com.josycom.mayorjay.flowoverstack.model.Answer
import com.josycom.mayorjay.flowoverstack.network.ApiService
import com.josycom.mayorjay.flowoverstack.util.AppConstants
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AnswerRepository @Inject constructor(private val apiService: ApiService) {

    var answerDataFlow: Flow<PagingData<Answer>>? = null

    fun init(questionId: Int, order: String, sortCondition: String, site: String, filter: String, siteKey: String) {
        answerDataFlow = Pager(PagingConfig(AppConstants.PAGE_SIZE, enablePlaceholders = false),
                pagingSourceFactory = {AnswerPagingSource(questionId, order, sortCondition, site, filter, siteKey, apiService)}).flow
    }
}